week3/Q1.py

```diff
@@ -0,0 +1,177 @@
+def get_graph(values, n, m):
+	'''
+	Builds a graph.
+
+	Args:
+		values: str, elements of the table, all rows in one line
+		n: int, number of columns
+		m: int, number of rows
+
+	Returns:
+		dict, includes two elements with keys 'nodes' (stores a dict where key is number of a node and value is node's value) and 'vertices' (stores a dict where key is a number of a node and value is a set of numbers of nodes to which it is connected)
+	'''
+	nodes = {i: values[i] for i in range(n*m)}
+	vertices = []
+	for i in range(n*m):
+		if n == 1 and m == 1:
+			vertices.append({})
+			break
+		elif n == 1 or m == 1:
+			if i == 0:
+				vertices.append({i + 1})
+			elif (i == m - 1 and n == 1) or (i == n - 1 and m == 1):
+				vertices.append({i - 1})
+			else:
+				vertices.append({i - 1, i + 1})
+		else:
+			#up left corner
+			if i == 0:
+				vertices.append({i + 1, i + n, i + n + 1})
+			#up right corner
+			elif i == n - 1:
+				vertices.append({i - 1, i + n, i + n - 1})
+			#down right corner
+			elif i == n*m - 1:
+				vertices.append({i - 1, i - n, i - n - 1})
+			#down left corner
+			elif i == n*(m - 1): 
+				vertices.append({i + 1, i - n, i - n + 1})
+			#down sight
+			elif i > n*(m - 1):
+				vertices.append({i - 1, i + 1, i - n, i - n - 1, i - n + 1})
+			#up sight
+			elif i < n:
+				vertices.append({i - 1, i + 1, i + n, i + n - 1, i + n + 1})
+			#left sight
+			elif i % n == 0:
+				vertices.append({i + 1, i - n, i + n, i - n + 1, i + n + 1})
+			#right sight
+			elif (i + 1) % n == 0:
+				vertices.append({i - 1, i - n, i + n, i - n - 1, i + n - 1})
+			#center
+			else:
+				vertices.append({i - 1, i + 1, i - n, i + n, i - n - 1, i - n + 1, i + n - 1, i + n + 1})
+
+	return {'nodes': nodes, 'vertices': vertices}
+
+
+def _get_prefixes_list(dictionary):
+	'''
+	Gets list of all possible prefixes in dictionary.
+
+	Args:
+		dictionary: set
+
+	Returns:
+		set, all possible prefixes
+	'''
+	prefixes = set()
+	for word in dictionary:
+		for i, letter in enumerate(word):
+			prefixes.add(word[:i + 1])
+	return prefixes
+
+
+def _isWord(line, dictionary):
+	'''
+	Checks if sequence of elements is a real word.
+
+	Args:
+		line: str, sequence of elements
+		dictionary: set
+
+	Returns:
+		boolean
+	'''
+	if line in dictionary:
+		return True
+	else:
+		return False
+
+
+def _isPrefix(line, prefixes):
+	'''
+	Checks if sequence of elements is a possible prefix.
+
+	Args:
+		line: str, sequence of elements
+		prefixes: set
+
+	Returns:
+		boolean
+	'''
+	if line in prefixes:
+		return True
+	else:
+		return False
+
```

Comment:
An alternative (which I think was the assignment author intention), would be to define a class:

```python
class Dictionary:
  def isWord(self, word):
     [...]
  def isPrefix(self, prefix):
     [...]
```